### DISCUSSION TITLE
How to make voting rights expire?

### DISCUSSION THREAD

**Discussion by IvanPayan** [snapshot-labs/snapshot#3383] [Q&A] [answered]
For membership management. 

Example: A user subscribes 1 year to an association and gets a soulbond. This soulbond gives him the right to vote. After one year, he will have to renew his membership or lose his voting rights.

How could this use case be implemented?

**Comment by samwhosung** [comment, marked as answer]
Yes this is possible with [voting validation](https://snapshot.mirror.xyz/wFAGLoy1EKRFr34wQ8lYdrdcCoENeyrgjyEGl_DPn00). You could set up a [custom validation](https://github.com/snapshot-labs/snapshot-strategies/tree/master/src/validations) to check for an NFT and have a condition for the timestamp of its creation.

@ChaituVR @bonustrack any other ways to achieve this?

**Comment by IvanPayan** [comment]
Thank you so much for your answer @samuveth , will check that way :)
